moveit_ros/hybrid_planning/global_planner/global_planner_plugins/src/moveit_planning_pipeline.cpp

```diff
@@ -97,44 +97,86 @@ moveit_msgs::msg::MotionPlanResponse MoveItPlanningPipeline::plan(
     return response;
   }
 
-  // Process goal
-  if ((global_goal_handle->get_goal())->motion_sequence.items.size() > 1)
+  // If multiple waypoints were requested, assume we have a fully planned trajectory already.
+  // Just forward to the local planner.
+  // This works only for a JointConstraint type.
+  size_t num_waypoints = global_goal_handle->get_goal()->motion_sequence.items.size();
+  if (num_waypoints > 1)
```

Comment:
I would prefer that the user actively needs to configure this option. Something like a simple "allow pass-through" [True/False] flag